### PR TITLE
Imx8qm fixes2 alternative path

### PR DESCRIPTION
**Pull request by igoropaniuk**

### CODE REVIEW

**Review by MrCry0 — APPROVED** [review]
LGTM

**Comment by igoropaniuk** [comment]
@MrCry0 could you please review again, I've added additional commit. Thanks!

File: meta-lmp-bsp/recipes-bsp/u-boot/u-boot-fio/imx8qm-mek/lmp.cfg [hunk 1]
```diff
@@ -29,6 +29,7 @@ CONFIG_CMD_IMPORTENV=y
 CONFIG_CMD_EDITENV=y
 CONFIG_CMD_SAVEENV=y
 CONFIG_BOOTDELAY=-2
+# CONFIG_BOOTCOUNT_LIMIT is not set
```
Comment (MrCry0) [inline meta-lmp-bsp/recipes-bsp/u-boot/u-boot-fio/imx8qm-mek/lmp.cfg:32]: Could you add a sense of unsetting (about moving handling to boot.cmd) to this commit message? Thanks.

Comment (igoropaniuk) [reply meta-lmp-bsp/recipes-bsp/u-boot/u-boot-fio/imx8qm-mek/lmp.cfg:32]: it there already, no?
```
By default CONFIG_BOOTCOUNT_LIMIT is being set in U-Boot, so explicitly
unset it here.
```

**Review by Tim-Anderson — APPROVED** [review]
LGTM

File: meta-lmp-bsp/recipes-bsp/u-boot/u-boot-fio/imx8qm-mek/lmp.cfg (hunk 1 above)
Comment (MrCry0) [reply meta-lmp-bsp/recipes-bsp/u-boot/u-boot-fio/imx8qm-mek/lmp.cfg]: The commit is gone :)

**Comment by mike-scott** [comment]
@igoropaniuk This needs a rebase, but I've tested both the upgrade and rollback and it seems to be working now.  LGTM

File: meta-lmp-base/recipes-bsp/u-boot/u-boot-ostree-scr-fit/boot-common-alternative.cmd.in [hunk 2]
```diff
@@ -107,21 +138,32 @@ if test "${fiovb.is_secondary_boot}" = "0"; then
...
+
 				echo "${fio_msg} doing reset to boot newly updated images..."
 				run do_reboot
 			fi;
 		fi
 	fi
-	# That means that ak-lite confirmed succesfull boot, so we can now reset
-	# fiovb.bootupgrade_available variable
-	if test "${fiovb.bootupgrade_available}" = "1" && test "${fiovb.upgrade_available}" = "0"; then
+	# That means that ak-lite confirmed succesful boot, so we can now reset
```
Comment (mike-scott) [inline meta-lmp-base/recipes-bsp/u-boot/u-boot-ostree-scr-fit/boot-common-alternative.cmd.in:154]: nit: Technically it's `successful` here.

File: meta-lmp-base/recipes-bsp/u-boot/u-boot-ostree-scr-fit/boot-common-alternative.cmd.in [hunk 3]
```diff
@@ -145,7 +187,7 @@ setenv bootcmd_rollback 'if test -n "${kernel_image2}" && test "${fiovb.rollback
 if test "${fiovb.is_secondary_boot}" = "1"; then
 	# and it became corrupted after update, which means
 	# that we should rollback also
-	if test "${fiovb.bootupgrade_available}" = "1"; then
+	if test "${fiovb.bootupgrade_primary_updated}" = "1"; then
```
Comment (mike-scott) [inline meta-lmp-base/recipes-bsp/u-boot/u-boot-ostree-scr-fit/boot-common-alternative.cmd.in:190]: IMHO, we should check `upgrade_available` to determine if we `run rollback_setup` (both being rootfs focused).
I'll leave it up to you, tho.  I don't feel that strongly either way.

**Review by mike-scott — APPROVED** [review]
Few minor nits, but LGTM

**Pull request merged**